### ISSUE TITLE
Xcode 10 Beta 6 Compiling Issue 

### ISSUE THREAD

**Issue by ghost** [christophhagen/libsignal-protocol-swift#5]
Compiling the framework with Xcode 10 Beta 6 does not work. I'm getting the following error: `Command CompileSwiftSources failed with a nonzero exit code`.  In order to reproduce the error just download the framework and try to build it. 

How can I solve this issue?

**Comment by guarland** [comment]
Have you solved this problem?

**Comment by BaranEm** [comment]
No, I didn't solve the problem, but I'm going to solve it this week. 

Are you facing the same issue? I didn't try it out on the Xcode release from the apple store. Is it still there?

**Comment by BaranEm** [comment]
Found the solution: 
1. First navigate to "libsignal-protocol-swift" -> "Targets" -> "SignalModuleMap".
2. On "SignalModuleMap" navigate to "Build Phases" and expand "Run Script".
3. Replace the existing script with this one: 

`# This if-statement means we'll only run the main script if the SignalModuleMap directory doesn't exist
# Because otherwise the rest of the script causes a full recompile for anything where SignalProtocol is a dependency
# Do a "Clean Build Folder" to remove this directory and trigger the rest of the script to run
if [ -d "${BUILT_PRODUCTS_DIR}/SignalModuleMap" ]; then
echo "${BUILT_PRODUCTS_DIR}/SignalModuleMap directory already exists, not creating module map."
else

mkdir -p "${BUILT_PRODUCTS_DIR}/SignalModuleMap"
cat <<EOF > "${BUILT_PRODUCTS_DIR}/SignalModuleMap/module.modulemap"
module SignalModule [system] {
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/signal_protocol.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/key_helper.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/session_builder.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/session_cipher.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/protocol.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/group_session_builder.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/group_cipher.h"
    header "${SRCROOT}/libsignal-protocol-swift/libsignal-protocol-c/fingerprint.h"
    export *
}
EOF
fi

COMMON_CRYPTO_DIR="${SDKROOT}/usr/include/CommonCrypto"
if [ -f "${COMMON_CRYPTO_DIR}/module.modulemap" ]; then :; fi`

4. Select each target in the "Targets" list and navigate to "Build Settings". Enter "Swift Language Version" into the search field and change "Swift 4" to "Swift 4.2". Note that this must be done on every target in the list!
5. That's it. It should work now.

**Comment by DivakarSwift** [comment]
I have replaced phase script but It's still showing error in GroupClipher like No such module 'SignalModule' and not able to import in my ViewController

**Comment by BaranEm** [comment]
Sorry for the late reply. Later this day I'm going to make a pull request (never done it before). I'm going to make sure that the library compiles without any errors and works on my side. If you still get the error `No such module` you need to investigate further in it. I did fight with these warnings/errors to, but don't know how I solved them. Google for the error and try every solution you find (that's what I did). Make sure to create backups of your project in order to recover without any problems when you did mess something up in the project settings and don't know how to go back.
